src/test/java/uk/gov/hmcts/reform/cpo/auditlog/aop/ExpressionEvaluatorTest.java

```diff
@@ -0,0 +1,144 @@
+import org.springframework.expression.spel.SpelEvaluationException;
+import org.springframework.util.ReflectionUtils;
+import uk.gov.hmcts.reform.BaseTest;
+import uk.gov.hmcts.reform.cpo.domain.CasePaymentOrder;
+
+import java.lang.reflect.Method;
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.springframework.test.util.AssertionErrors.assertEquals;
+import static org.springframework.test.util.AssertionErrors.assertNull;
+
+class ExpressionEvaluatorTest implements BaseTest {
+
+    private ExpressionEvaluator evaluator;
+
+    @BeforeEach
+    void setUp() {
+        evaluator = new ExpressionEvaluator();
+    }
+
+    @Test
+    @DisplayName("should create valuation context")
+    void shouldCreateEvaluationContext() {
+
+        // GIVEN
+        String stringValue = "test";
+        int intValue = 100;
+        Method method = ReflectionUtils.findMethod(SampleMethods.class, "hello", String.class, Boolean.class);
+
+        // WHEN
+        EvaluationContext context = evaluator.createEvaluationContext(this, SampleMethods.class, method,
+                                                                      new Object[]{stringValue, intValue}
+        );
+
+        // THEN
+        assertEquals("Should match 1st arg on 'a0'",
+                     stringValue, context.lookupVariable("a0")
+        );
+        assertEquals("Should match 1st arg on 'p0'",
+                     stringValue, context.lookupVariable("p0")
+        );
+        assertEquals("Should match 1st arg on name",
+                     stringValue, context.lookupVariable("foo")
+        );
+
+        assertEquals("Should match 2nd arg on 'a1'",
+                     intValue, context.lookupVariable("a1")
+        );
+        assertEquals("Should match 2nd arg on 'p1'",
+                     intValue, context.lookupVariable("p1")
+        );
+        assertEquals("Should match 1st arg on name",
+                     intValue, context.lookupVariable("flag")
+        );
+
+        assertNull("Should return null when 3rd arg not found via index 'a2'",
+                     context.lookupVariable("a2")
+        );
+        assertNull("Should return null when 3rd arg not found via index 'p2'",
+                     context.lookupVariable("p2")
+        );
+    }
+
+    @Test
+    @DisplayName("should parse bean expression")
+    void shouldParseBeanExpressions() {
+
+        // GIVEN
+        String stringValue = "test";
+        CasePaymentOrder casePaymentOrder = createCasePaymentOrder();
+        Method method = ReflectionUtils.findMethod(SampleMethods.class, "hello", String.class, CasePaymentOrder.class);
+
+        EvaluationContext context = evaluator.createEvaluationContext(this, SampleMethods.class, method,
+                                                                      new Object[]{stringValue, casePaymentOrder}
+        );
+        assert method != null;
```

Comment:
Should this be a Junit assertion ?

Reply:
Resolved in 8fcbda9faaefab2c281ae0d6bc0cee8f12cbcac2. Sorry was trying to avoid some Intelij warnings. I have now switched it to use `Objects.requireNonNull(method)` as it possibly more discrete / less confusing when placed alongside tests.   Thanks.